src/vivarium_conic_lsff/components/observers.py

```diff
@@ -521,3 +522,89 @@ def get_lbwsg_stats(self, pop):
     def metrics(self, index, metrics):
         metrics.update(self.results)
         return metrics
+
+
+class HemoglobinLevelObserver():
+
+    @property
+    def name(self):
+        return project_globals.HEMOGLOBIN_OBSERVER
+
+
+    def setup(self, builder):
+        self.hemoglobin = builder.value.get_value(f'{project_globals.IRON_DEFICIENCY_MODEL_NAME}.exposure')
+        self.iron_responsive = builder.value.get_value('iron_responsive')
+
+        self.record_ages = [0.5, 1, 2, 3]
+        self.population_view = builder.population.get_view(['age',
+                                                            project_globals.IRON_COVERAGE_START_AGE_COLUMN],
+                                                           query='alive == "alive"')
+        self.results = self.get_results_template()
+
+        builder.event.register_listener('collect_metrics', self.on_collect_metrics)
+        builder.value.register_value_modifier('metrics', self.metrics)
+
+    def on_collect_metrics(self, event):
+        pop = self.population_view.get(event.index)
+        for age in self.record_ages:
+            pop_age = pop[(age <= pop.age) & (pop.age < age + to_years(event.step_size))]
+
+            responsive = self.iron_responsive(pop_age.index)
+            idx_resp = responsive[responsive].index
+            idx_non_resp = responsive[~responsive].index
+
+            idx_covered = pop_age.loc[(pop_age.age > 0.5)
```

Comment:
Calling this `idx_covered` seems wrong, since it's excluding covered but non-responsive children. I think you should drop line 558 so that you get the proper 2x2 grid of covered vs responsive.

Reply:
> >  I think you should drop line 558 so that you get the proper 2x2 grid of covered vs responsive.
Done, thanks.